include/oneapi/dpl/internal/random_impl/counter_based_engines_stuff.h

```diff
@@ -0,0 +1,98 @@
+// -*- C++ -*-
+//===-- counter_based_engines_stuff.h -------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+//
+// Abstract:
+//
+// Header file provides stuff functionality for counter-based RNG engines.
+//
+//   word_mask<Uint, w> - the Uint with the low w bits set
+
+
+#ifndef _ONEDPL_CTR_ENGINES_STUFF_H
+#define _ONEDPL_CTR_ENGINES_STUFF_H
+
+#include <concepts>
+#include <iterator>
+#include <utility>
+#include <tuple>
+#include <limits>
+
+namespace oneapi
```

Comment:
Suggestion: to move it under oneapi::dpl::detail::random:: namespace. The functionality is not needed in the other parts

Reply:
Discussed offline to make it similar to `include/oneapi/dpl/internal/random_impl/random_common.h`
![image](https://github.com/user-attachments/assets/3bb4fa04-43f2-442d-b1c0-ae69088ea56b)
done in https://github.com/oneapi-src/oneDPL/pull/1785/commits/50c586c904501a28a5bcd6d440bbb260ba34cea8